src/app/Profile/page.tsx

```diff
@@ -1,3 +1,224 @@
+    { id: "3", name: "Item 3", isLocked: false },
+    { id: "4", name: "Item 4", isLocked: true },
+    { id: "5", name: "Item 5", isLocked: true },
+    { id: "6", name: "Item 6", isLocked: false },
+    { id: "7", name: "Item 7", isLocked: false },
+    { id: "8", name: "Item 8", isLocked: false },
+    { id: "9", name: "Item 9", isLocked: true },
+    { id: "10", name: "Item 10", isLocked: true },
+    { id: "11", name: "Item 11", isLocked: true },
+    { id: "12", name: "Item 12", isLocked: true }
+  ];
+
+  return (
+    <div className="fixed top-0 left-0 w-screen h-screen bg-white bg-opacity-60 z-50">
+      <div className="fixed bottom-0 bg-white bg-opacity-80 max-w-sm w-full h-[50%] shadow-md mx-auto left-0 right-0 rounded-t-3xl overflow-hidden">
+        <div className="relative flex justify-center items-center h-16 bg-gradient-to-t from-[#D2CAFF] to-[#092B44]">
+          <p className="text-[#ECF0F6] text-2xl font-semibold">Choose your artifacts</p>
+          <button
+            onClick={onClose}
+            className="absolute right-5"
+          >
+            <img
+              src="./profile/cross.webp"
+              alt="X"
+              className="w-full h-full"
+            />
+          </button>
+        </div>
+
+        <div className="p-4 text-center">
+          <ItemsGrid items={items} />
+        </div>
+      </div>
+    </div>
+  );
+};
+
+type Item = {
+  id: string;
+  name: string;
+  isLocked: boolean;
+};
+
+function ItemsGrid({ items }: { items: Item[] }) {
+  return (
+    <div className="grid grid-cols-3 w-full justify-center justify-items-center bg-[#ECF0F6]/80 h-72"
+        style={{
+        overflow: "auto", 
+        scrollbarWidth: "none", 
+        msOverflowStyle: "none", 
+        WebkitOverflowScrolling: "touch", 
+        overflowY: "scroll", 
+      }}>
+      {items.map((item) => (
+        <div
+          key={item.id}
+          className={`flex flex-col items-center justify-center bg-[#7D7D7D] w-24 h-28 my-2.5 mx-2`}
+          style={{ minHeight: "100px" }}
+        >
+          {item.isLocked && <img
+            src="/profile/locked.webp"
+            alt="Locked"
+            className="w-[30%] h-[30%] object-contain"
+          />}
+        </div>
+      ))}
+    </div>
+  );
+};
+
+
 export default function Profile() {
-  return <h1>Hi this is Profile page</h1>;
+  const [name, setName] = useState("")
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [selectedArtifact, setSelectedArtifact] = useState<number | null>(null);
+
+  const [selectedParts, setSelectedParts] = useState<{
+      skin: string | null;
+      hair: string | null;
+      eyebrow: string | null;
+      eye: string | null;
+      nose: string | null;
+      mouth: string | null;
+      shirt: string | null;
+      clothes: string | null;
+      pant: string | null;
+      shoes: string | null;
+    }>(() => {
+      const savedAvatar = localStorage.getItem("selectedParts");
```

Comment:
add type for `savedAvatar` because "Unsafe return of a value of type `any`." in line 110